### PR TITLE
Membership reference constraints

### PR DESCRIPTION
**Pull request by matslats**
removed a constraint from OgMembershipReferenceItem which was expecting a list. Not sure whether that constraint is needed elsewhere. Probably not

### CODE REVIEW

**Comment by matslats** [comment]
it says the service entity_field,manager doesn't exist, which is wrong. It does!

**Comment by pfrenssen** [comment]
I hit this bug as well. This bug is triggered when adding a group entity to the group audience field. When the ValidReferenceConstraintValidator tries to validate the input it thinks it is passed an EntityReferenceFieldItemList and tries to iterate over the values in the list. This obviously fails, and it throws this error:

> Error: Call to undefined method Drupal\Core\TypedData\Plugin\DataType\IntegerData::isEmpty() in Drupal\Core\Entity\Plugin\Validation\Constraint\ValidReferenceConstraintValidator->validate() (line 78 of core/lib/Drupal/Core/Entity/Plugin/Validation/Constraint/ValidReferenceConstraintValidator.php).

File: src/OgAccess.php [hunk 1]
```diff
@@ -191,7 +191,8 @@ public static function userAccessEntity($operation, EntityInterface $entity, Acc
       foreach ($entity_groups as $groups) {
         foreach ($groups as $group) {
           $user_access = static::userAccess($group, $operation, $user);
-          if ($user_access->isAllowed()) {
+          if (!$user_access->isForbidden()) {
+            //this covers allowed and neutral
             return $user_access->addCacheTags($cache_tags);
```
Comment (pfrenssen) [inline src/OgAccess.php:196]: The coding standards are very specific for comments, there should be a space after the `//`, the first word should be capitalized, and the sentence should end in a period. Painful, I know ;)

File: tests/src/Unit/OgAccessEntityTestBase.php [hunk 2]
```diff
@@ -44,9 +44,9 @@ public function setup() {
 
     $this->groupManager->isGroup($entity_type_id, $bundle)->willReturn(FALSE);
 
-    $entity_manager = $this->prophesize(EntityManagerInterface::class);
-    $entity_manager->getFieldDefinitions($entity_type_id, $bundle)->willReturn([$field_definition->reveal()]);
-    \Drupal::getContainer()->set('entity.manager', $entity_manager->reveal());
+    $entity_field_manager = $this->prophesize(EntityFieldManagerInterface::class);
+    $entity_field_manager->getFieldDefinitions($entity_type_id, $bundle)->willReturn([$field_definition->reveal()]);
+    \Drupal::getContainer()->set('entity.manager', $entity_field_manager->reveal());
```
Comment (pfrenssen) [inline tests/src/Unit/OgAccessEntityTestBase.php:49]: To fix the failures, try changing this line to

`\Drupal::getContainer()->set('entity_field.manager', $entity_field_manager->reveal());`

**Comment by pfrenssen** [comment]
I created a new PR that contains only this fix, not the unrelated commits: #138.

**Pull request closed without merging**